contracts/tokenomics/generator/src/contract.rs

```diff
@@ -1450,10 +1450,21 @@ fn init_proxy_rewards_holder(
     Ok(msg)
 }
 
+/// Adds a new proxy contract to the list
+fn update_reward_proxies(deps: DepsMut, proxy: &Addr) -> Result<(), ContractError> {
+    let mut all_proxies = REWARD_PROXIES_LIST.load(deps.storage)?;
+    if !all_proxies.contains(proxy) {
+        all_proxies.push(proxy.clone());
+        REWARD_PROXIES_LIST.save(deps.storage, &all_proxies)?;
```

Comment:
We have the proxies list [already](https://github.com/astroport-fi/astroport-core/blob/1b94b050c6d1ca8315f14df7a16e3b963fd8f388/contracts/tokenomics/generator/src/state.rs#L152). I suggest to use `PROXY_REWARD_ASSET.keys()` and remove REWARD_PROXIES_LIST and this function.

Reply:
Yes, but ```keys``` method will deserialize values [too](https://github.com/CosmWasm/cw-plus/blob/580e555dd091fd7d80c0e291f4c65fd418e87981/packages/storage-plus/src/prefix.rs#L179), thus limiting maximum possible elements in the array due to query gas limitation. Something similar is with storing this array into the config

Reply:
Ok, will do for code clarity and gas saving at the beginning